tests/drivers/adc/adc_dma/boards/frdm_k64f.overlay

```diff
@@ -10,3 +10,9 @@
 	high-speed;
 	periodic-trigger;
 };
+
+&edma0 {
+	dma-buf-alignment = <32>;
```

Comment:
This should be in the soc dtsi not a test overlay so other people can use it

Reply:
@teburd , for nxp edma, the buffer alignment is kind of variable 32 bit is the maximum requirement, see hal code below
```
assert((srcWidth != 0U) && (srcWidth <= 32U) && ((srcWidth & (srcWidth - 1U)) == 0U));
    assert((destWidth != 0U) && (destWidth <= 32U) && ((destWidth & (destWidth - 1U)) == 0U));
    assert((transferBytes % bytesEachRequest) == 0U);
    assert((((uint32_t)(uint32_t *)srcAddr) % srcWidth) == 0U);
    assert((((uint32_t)(uint32_t *)destAddr) % destWidth) == 0U);
``` 
it requires the destAddr be at most 32 and the actuall alignment is rounded by destWidth, so this is not a fixed number, do we still need set a fix number in soc level?

Reply:
Ok I see the problem now, yes so in this case dma-buf-alignment isn't really a static property of the dma controller, and that's fine. It shouldn't use that property at all then.

However... a magic alignment of 32 bytes isn't that great either in this case. Is 32 byte alignment always going to be used for NXP eDMA based devices with ADC? Perhaps adding an adc-buf-alignment property in this case is more appropriate.

Either way this information of needing a specific alignment for a buffer for this device should be encoded in some place besides in a test case. Removing my -1 as I think this needs to be decided by the ADC maintainer not by myself though.

Reply:
@teburd yes dma-buf-alignment = <4>, I updated.